charmcraft/commands/build.py

```diff
@@ -191,14 +193,29 @@ def run(self, bases_indices: Optional[List[int]] = None) -> List[str]:
                     continue
 
                 for j, build_on in enumerate(bases_config.build_on):
-                    matches, reason = check_if_base_matches_host(build_on)
+                    if is_charmcraft_running_in_managed_mode():
+                        matches, reason = check_if_base_matches_host(build_on)
+                    else:
+                        matches, reason = is_base_providable(build_on)
+
                     if matches:
                         logger.debug(
                             "Building for 'bases[%d]' as host matches 'build-on[%d]'.",
                             i,
                             j,
                         )
-                        charm_name = self.build_charm(bases_config)
+                        if is_charmcraft_running_in_managed_mode():
+                            charm_name = self.build_charm(bases_config)
+                        else:
+                            with launched_environment(
+                                charm_name=self.metadata.name,
+                                project_path=self.charmdir,
+                                base=build_on,
+                                bases_index=i,
```

Comment:
personal preference comment, and I think I made it in the past in a different PR, but i and j should only be used for simple iterations, it's still used 10 lines below and hard to keep track of

Reply:
Good point, will fix!

Reply:
Done